### DISCUSSION TITLE
Provide the user with a way to customize the ffmpeg command used to render their video

### DISCUSSION THREAD

**Discussion by WampyCakes** [ManimCommunity/manim#2785] [Suggestions and Proposals]
## Description of proposed feature
#1144 introduces a change to the ffmpeg command used to create a GIF. While it seems to handle every scene I've tested well, I am sure there will be scenes that benefit from different settings (such as a different dithering). I think that we should either provide a way to set in a script `config["ffmpeg_command"]` or at least providing through the config more ways to control GIF (read: ffmpeg) flags. 

## How can the new feature be used?
Examples of these are adjusting the current `palletegen` and `palleteuse` in #1144 or allowing the user to add flags like the following to reduce file size:
```
scale=320:-1:flags=lanczos
```

## Additional comments
I am not sure of the exact form this should take regarding the dynamic properties used in the ffmpeg command. Perhaps just passing an array with placeholders and then executing exactly what the user provides would be enough. (Like interpreting `file_list` as the literal file_list variable)

**Comment by leotrs** [comment]
I am not opposed to this, but it sounds like it will require some non-trivial refactoring of scene_file_writer.

Dynamic properties can already be achieved for paths saved in config (see ManimConfig.get_dir).

**Comment by WampyCakes** [comment]
This customizability would also fix #786.

**Comment by blackedout01** [comment]
I have an example of why this proposal is useful for plain video output. Currently, the ffmpeg setting crf seems to be hard coded into manim, like in `manim/scene/scene_file_writer.py` where the method `open_partial_movie_stream` contains:
```python
av_options = {
    "an": "1",  # ffmpeg: -an, no audio
    "crf": "23",  # ffmpeg: -crf, constant rate factor (improved bitrate)
}
```

Changing this value can avoid certain video artifacts in the output. The following code produces compression artifacts because of crf being set to 23.

```python
class OpacityIssue(Scene):
    def construct(self):
        r = Text("hello")
        self.play(FadeIn(r))
        self.wait()
        self.play(r.animate.set_opacity(0.5))
        self.wait()
```

https://github.com/user-attachments/assets/dcb7e5b2-6c41-4d6a-848e-cf2acdbc7310

Generated using Manim Community v0.19.0 and screen caputured a zoomed in version in QuickTime on macOS.
It might seem not very noticable without such a large zoom, but I noticed these artifacts just while working on some tex opacity animations. The delayed pixel change caused the output to look like the animation has a stutter at the end.
The issue is not present when using a crf value of 10.

My proposal is a new option `video_crf` that can then be set in the manim config or as an additional command line option.
